Intersect.Server/Entities/Player.cs

```diff
@@ -6470,52 +6470,64 @@ CommonEventTrigger trigger
             }
         }
 
-        public override int CanMove(Direction moveDir)
+        public override bool CanMoveInDirection(Direction direction) => CanMoveInDirection(direction, out _, out _);
+
+        public override bool CanMoveInDirection(Direction direction, 
+            out MovementBlockerType blockerType, 
+            out EntityType entityType)
         {
             //If crafting or locked by event return blocked
             if (OpenCraftingTableId != default && CraftingState != default)
             {
-                return -5;
+                blockerType = MovementBlockerType.OutOfBounds;
+                entityType = EntityType.GlobalEntity;
+                return false;
             }
 
             foreach (var evt in EventLookup)
             {
                 if (evt.Value.HoldingPlayer)
                 {
-                    return -5;
+                    blockerType = MovementBlockerType.OutOfBounds;
+                    entityType = EntityType.GlobalEntity;
+                    return false;
                 }
             }
-
-            return base.CanMove(moveDir);
+            
+            return base.CanMoveInDirection(direction, out blockerType, out entityType);
         }
 
-        protected override int IsTileWalkable(MapController map, int x, int y, int z)
+        protected override bool IsTileWalkable(MapController map, int x, int y, int z)
         {
-            if (base.IsTileWalkable(map, x, y, z) == -1)
+            if (!base.IsTileWalkable(map, x, y, z))
             {
-                foreach (var evt in EventLookup)
+                return false;
+            }
+
+            foreach (var evt in EventLookup)
+            {
+                if (evt.Value.PageInstance == null)
                 {
-                    if (evt.Value.PageInstance != null)
-                    {
-                        var instance = evt.Value.PageInstance;
-                        if (instance.GlobalClone != null)
-                        {
-                            instance = instance.GlobalClone;
-                        }
+                    continue;
+                }
 
-                        if (instance.Map == map &&
-                            instance.X == x &&
-                            instance.Y == y &&
-                            instance.Z == z &&
-                            !instance.Passable)
-                        {
-                            return (int) EntityType.Event;
-                        }
-                    }
+                var instance = evt.Value.PageInstance;
+                if (instance.GlobalClone != null)
+                {
+                    instance = instance.GlobalClone;
+                }
+
+                if (instance.Map == map &&
+                    instance.X == x &&
+                    instance.Y == y &&
+                    instance.Z == z &&
+                    !instance.Passable)
+                {
+                    return false;
```

Comment:
This is losing information. The method returned the blocker before, and this does not return a blocker.